pkg/platform/update-api/update_api.go

```diff
@@ -0,0 +1,134 @@
+package update_api
+
+import (
+	"encoding/json"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+const bottlerocketApiSock = "/run/api.sock"
+
+type updateState = string
+
+const (
+	Idle      updateState = "Idle"
+	Available updateState = "Available"
+	Staged    updateState = "Staged"
+	Ready     updateState = "Ready"
+)
+
+type UpdateImage struct {
+	Arch    string `json:"arch"`
+	Version string `json:"version"`
+	Variant string `json:"variant"`
+}
+
+func (ui *UpdateImage) Identifier() interface{} {
+	return ui.Version
+}
+
+type StagedImage struct {
+	Image      UpdateImage `json:"image"`
+	NextToBoot bool        `json:"next_to_boot"`
+}
+
+type updateCommand = string
+
+const (
+	refresh    updateCommand = "refresh"
+	prepare    updateCommand = "prepare"
+	activate   updateCommand = "activate"
+	deactivate updateCommand = "deactivate"
+)
+
+type commandStatus = string
+
+const (
+	Success commandStatus = "Success"
+	Failed  commandStatus = "Failed"
+	Unknown commandStatus = "Unknown"
+)
+
+type CommandResult struct {
+	CmdType    updateCommand `json:"cmd_type"`
+	CmdStatus  commandStatus `json:"cmd_status"`
+	Timestamp  string        `json:"timestamp"`
+	ExitStatus *int32        `json:"exit_status"`
+	Stderr     *string       `json:"stderr"`
+}
+
+type UpdateStatus struct {
+	UpdateState       updateState    `json:"update_state"`
+	AvailableUpdates  []string       `json:"available_updates"`
+	ChosenUpdate      *UpdateImage   `json:"chosen_update"`
+	ActivePartition   *StagedImage   `json:"active_partition"`
+	StagingPartition  *StagedImage   `json:"staging_partition"`
+	MostRecentCommand *CommandResult `json:"most_recent_command"`
+}
+
+// getUpdateStatus repeatedly tries to GET /updates/status until a status is returned
+// We only retry the request if the response was a 423 Locked (indicates that the update status is not ready)
+func getUpdateStatus(httpClient http.Client) (*UpdateStatus, error) {
+	var response *http.Response
+	var err error
+	var attempts = 0
+	// Retry up to 5 times in case the Update API is busy; Waiting 10 seconds between each attempt.
+	for attempts < 5 {
+		response, err = httpClient.Get("http://unix" + "/updates/status")
+		if err != nil {
+			errors.New("error when trying to GET '/updates/status'")
+			return nil, err
+		}
+		// Response OK, break out of loop
+		if response.StatusCode == 204 {
```

Comment:
You'll probably want to accept any 2xx code.


```suggestion
		if response.StatusCode >= 200 && response.StatusCode < 300 {
```